### ISSUE TITLE
Complete auth should work

### ISSUE THREAD

**Issue by jazeee** [kahmali/meteor-restivus#98]
I was looking at the code and testing functionality, and realized that the "Complete auth" option does not work.

From my tests, Complete Auth option does not work.

I believe it is a simple issue with route.coffee line 159:
https://github.com/kahmali/meteor-restivus/blob/devel/lib/route.coffee#L159

The line should be:

    auth = @api.config.auth.user?(endpointContext)  #without the .call 

I also believe the documentation should describe the function signature, and what it expects as a response such as:

    Restivus.configure 
      auth: 
        user: (endpointContext) ->
          token = endpointContext.request.headers["x-auth-token"]
          ... Do something with token and extract your user...
          if notAllowed(user) throw new Meteor.Error("Unauthorized")
          return user

**Comment by jazeee** [comment]
Sorry, I misunderstood how the Complete Auth works. I do believe a little more documentation will help with understanding.

**Comment by kahmali** [comment]
I agree that the auth section of the docs needs some improvement. I made a small update for #91, but the custom auth needs to be explained there as well. I'll keep this open until it's updated.
